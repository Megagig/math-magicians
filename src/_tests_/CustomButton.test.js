import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom/extend-expect';
import Buttons from '../components/CustomButton';

describe('Buttons component', () => {
  test('renders button with correct value and class name', () => {
    const handleClick = jest.fn();
    const value = 'Click Me';
    const className = 'custom-button';

    const { getByText } = render(
      <Buttons value={value} className={className} handleClick={handleClick} />
    );

    const buttonElement = getByText(value);
    expect(buttonElement).toBeInTheDocument();
    expect(buttonElement).toHaveClass(className);
  });

  test('calls handleClick function when clicked', () => {
    const handleClick = jest.fn();
    const value = 'Click Me';
    const className = 'custom-button';

    const { getByText } = render(
      <Buttons value={value} className={className} handleClick={handleClick} />
    );

    const buttonElement = getByText(value);
    fireEvent.click(buttonElement);

    expect(handleClick).toHaveBeenCalledTimes(1);
    expect(handleClick).toHaveBeenCalledWith(value);
  });
});

describe('Buttons snapshot', () => {
  test('matches the snapshot', () => {
    const handleClick = jest.fn();
    const value = 'Click Me';
    const className = 'custom-button';

    const tree = renderer
      .create(
        <Buttons
          value={value}
          className={className}
          handleClick={handleClick}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
