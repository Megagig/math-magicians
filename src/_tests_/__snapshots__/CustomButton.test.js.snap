// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Buttons snapshot matches the snapshot 1`] = `
<button
  className="btn custom-button"
  onClick={[Function]}
  type="button"
>
  Click Me
</button>
`;
